### DISCUSSION TITLE
RNN and Tranformers tabular data

### DISCUSSION THREAD

**Discussion by cornuch** [Lightning-AI/dl-fundamentals#55] [General discussions]
Is it possible to use RNN and/or transformers in a case where we have tabular data using time but if we have not a regular step in time (in order to predict a value in a future time) ?

**Comment by rasbt** [comment]
Transformers can definitely be used for tabular data

- TabPFN: A Transformer That Solves Small Tabular Classification Problems in a Second, https://arxiv.org/abs/2207.01848
- TabTransformer: Tabular Data Modeling Using Contextual Embeddings, https://arxiv.org/abs/2012.06678

and others.

And RNNs are particularly popular for time series problems. (But some transformers exist for that purpose as well).

I am not sure what you mean by

> if we have not a regular step in time

could you provide an example to illustrate it?

**Comment by cornuch** [comment]
Thank you 


De: "Sebastian Raschka" ***@***.***> 
À: "Lightning-AI/dl-fundamentals" ***@***.***> 
Cc: "CHRISTELLE CORNU" ***@***.***>, "Author" ***@***.***> 
Envoyé: Dimanche 21 Mai 2023 16:02:51 
Objet: Re: [Lightning-AI/dl-fundamentals] RNN and Tranformers tabular data (Discussion #55) 





Transformers can definitely be used for tabular data 

    * TabPFN: A Transformer That Solves Small Tabular Classification Problems in a Second, [ https://arxiv.org/abs/2207.01848 | https://arxiv.org/abs/2207.01848 ] 
    * TabTransformer: Tabular Data Modeling Using Contextual Embeddings, [ https://arxiv.org/abs/2012.06678 | https://arxiv.org/abs/2012.06678 ] 


and others. 

And RNNs are particularly popular for time series problems. (But some transformers exist for that purpose as well). 

I am not sure what you mean by 



if we have not a regular step in time 



could you provide an example to illustrate it?
